src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/DynatraceMetricsExporterTests.cs

```diff
@@ -0,0 +1,181 @@
+            mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.Is<HttpRequestMessage>(req => !req.Headers.Contains("Api-Token")), ItExpr.IsAny<CancellationToken>());
+            mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.Is<HttpRequestMessage>(req => req.Headers.Contains("User-Agent")), ItExpr.IsAny<CancellationToken>());
+            mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.Is<HttpRequestMessage>(req => req.Headers.GetValues("User-Agent").Count() == 1), ItExpr.IsAny<CancellationToken>());
+            mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.Is<HttpRequestMessage>(req => req.Headers.GetValues("User-Agent").First() == "opentelemetry-metric-dotnet"), ItExpr.IsAny<CancellationToken>());
+            mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.Is<HttpRequestMessage>(req => req.Content.ReadAsStringAsync().Result == "namespace1.metric1,label1=value1,label2=value2,dt.metrics.source=opentelemetry count,delta=100 1604660628881" + Environment.NewLine), ItExpr.IsAny<CancellationToken>());
+        }
+
+        [Fact]
+        public async void TestUriAndToken()
+        {
+            var mockMessageHandler = new Mock<HttpMessageHandler>();
+            mockMessageHandler.Protected()
+            .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+            .ReturnsAsync(new HttpResponseMessage
+            {
+                StatusCode = HttpStatusCode.OK,
+                Content = new StringContent("test")
+            });
+
+            var exporter = new DynatraceMetricsExporter(new DynatraceExporterOptions { Url = "http://my.url", ApiToken = "test-token" }, null, new HttpClient(mockMessageHandler.Object));
+
+            var result = await exporter.ExportAsync(new List<Metric> { CreateMetric() }, CancellationToken.None);
+
+            mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.Is<HttpRequestMessage>(req => req.RequestUri.AbsoluteUri == "http://my.url/"), ItExpr.IsAny<CancellationToken>());
+            mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.Is<HttpRequestMessage>(req => req.Headers.Contains("Authorization")), ItExpr.IsAny<CancellationToken>());
+            mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.Is<HttpRequestMessage>(req => req.Headers.GetValues("Authorization").Count() == 1), ItExpr.IsAny<CancellationToken>());
+            mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.Is<HttpRequestMessage>(req => req.Headers.GetValues("Authorization").First() == "Api-Token test-token"), ItExpr.IsAny<CancellationToken>());
```

Comment:
If you want to get fancy and avoid the long, repetitive lines, another option is to use [Callback](https://github.com/Moq/moq4/wiki/Quickstart#callbacks) and save the passed in `HttpRequestMessage` and assert on that instead: Like:

```
// this var will hold the actual passed in params
HttpRequestMessage actualRequestMessage = null;
 
mockMessageHandler.Protected()
.Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
.ReturnsAsync(new HttpResponseMessage
{
	StatusCode = HttpStatusCode.OK,
	Content = new StringContent("test")
})
.Callback((HttpRequestMessage r, CancellationToken _) => actualRequestMessage = r);
 ```
 
 Then you can assert normally 
 
 ```
// assert it was called once
mockMessageHandler.Protected().Verify("SendAsync", Times.Exactly(1), ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>());

Assert.Equal(HttpMethod.Post, actualRequestMessage.Method);
Assert.False(actualRequestMessage.Headers.Contains("Api-Token"));
...
 ```

Reply:
:exploding_head: That's pretty neat, NGL.